frontend/src/utils/showToastOnError.ts

```diff
@@ -0,0 +1,31 @@
+import type { ApiError } from "../client"
+
+interface ErrorDetail {
+  msg: string
+}
+
+interface ApiErrorBody {
+  detail?: string | ErrorDetail[]
+}
+
+// These variables are not used, they are just type definition
+/* eslint-disable */
+type ShowToastFunction = (
+  title: string,
+  description: string,
+  status: "success" | "error",
+) => void
+/* eslint-enable */
+
+export const handleError = (err: ApiError, showToast: ShowToastFunction) => {
```

Comment:
I'm wondering if there is a field on the `ApiError` class such as `statusText` or something else that has this information. If not, maybe we need to adjust the backend. I would think it wouldn't be necessary to do all of this to get the error message

Reply:
This utility function can definitely be revisited in future PR. I added these type definitions to pass the codacy error, the implementation detail is still the same as the original.